src/LiveComponent/src/LiveCollectionTrait.php

```diff
@@ -46,6 +46,70 @@ public function removeCollectionItem(PropertyAccessorInterface $propertyAccessor
         $propertyAccessor->setValue($this->formValues, $propertyPath, $data);
     }
 
+    #[LiveAction]
+    public function moveCollectionItemUp(PropertyAccessorInterface $propertyAccessor, #[LiveArg] string $name, #[LiveArg] int $index): void
+    {
+        if ($index <= 0) {
+            return;
+        }
+
+        $propertyPath = $this->fieldNameToPropertyPath($name, $this->formName);
+        $data = $propertyAccessor->getValue($this->formValues, $propertyPath);
+
+        if (!\is_array($data) || !isset($data[$index]) || !isset($data[$index - 1])) {
+            return;
+        }
+
+        $formConfig = $this->getForm()->get($name)->getConfig();
+        $orderPropertyPath = $formConfig->getOption('order_property_path');
+
+        if (null !== $orderPropertyPath) {
+            // Swap positions using the specified order property
+            $prevItemOrder = $propertyAccessor->getValue($data[$index - 1], $orderPropertyPath);
+            $currentItemOrder = $propertyAccessor->getValue($data[$index], $orderPropertyPath);
+
+            $propertyAccessor->setValue($data[$index], $orderPropertyPath, $prevItemOrder);
+            $propertyAccessor->setValue($data[$index - 1], $orderPropertyPath, $currentItemOrder);
+        } else {
+            // Legacy behavior - simple array swap (with warning about limitations)
+            $temp = $data[$index - 1];
+            $data[$index - 1] = $data[$index];
+            $data[$index] = $temp;
+        }
+
+        $propertyAccessor->setValue($this->formValues, $propertyPath, $data);
+    }
+
+    #[LiveAction]
+    public function moveCollectionItemDown(PropertyAccessorInterface $propertyAccessor, #[LiveArg] string $name, #[LiveArg] int $index): void
+    {
+        $propertyPath = $this->fieldNameToPropertyPath($name, $this->formName);
+        $data = $propertyAccessor->getValue($this->formValues, $propertyPath);
+
+        if (!\is_array($data) || !isset($data[$index]) || !isset($data[$index + 1])) {
+            return;
+        }
```

Comment:
This is not compatible with the add/remove behaviour for now..

The add set the new value as the max of the current ones + 1, but the remove simply unset it.

So if you start with

0 A
1 B
2 C

then delete B

0 A
2 C

If you move up A or move down C you will not find the previous / next item.